plugin/log/log.go

```diff
@@ -28,42 +28,43 @@ type Logger struct {
 func (l Logger) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
 	state := request.Request{W: w, Req: r}
 	for _, rule := range l.Rules {
-		if !plugin.Name(rule.NameScope).Matches(state.Name()) {
```

Comment:
Would initiating multiple rules with the new names make sense here? We mean none of this code needs to change

Reply:
> initiating multiple rules with the new names

is that means use multi log? such as:

```
log a.com
log b.com
```

That is a workaround of this situation, but it needs a format and a class for each name, which causes more memory. that's why i send this PR.

Reply:
1. is the better way

The map in the rule is a bit unfortunate (and causes most memory). Looks like an easy thing to remove/change.